src/sagemaker/jumpstart/artifacts/kwargs.py

```diff
@@ -71,13 +73,17 @@ def _retrieve_model_init_kwargs(
         tolerate_vulnerable_model=tolerate_vulnerable_model,
         tolerate_deprecated_model=tolerate_deprecated_model,
         sagemaker_session=sagemaker_session,
+        model_type=model_type,
     )
 
     kwargs = deepcopy(model_specs.model_kwargs)
 
     if model_specs.inference_enable_network_isolation is not None:
         kwargs.update({"enable_network_isolation": model_specs.inference_enable_network_isolation})
 
+    if model_type == JumpStartModelType.PROPRIETARY:
+        kwargs.update({"enable_network_isolation": True})
```

Comment:
question: is this overwriting a potentially overwriting the argument `enable_network_isolation=False` explicitly set by the customer?

If yes, we shouldn't, we should raise a `ValueError` with an error message explaining that network isolation is required for proprietary models.

Reply:
Thanks, added these for the discussion from bug bash that currently `enable_network_isolation` is set to False by default because there's no such entry in metadata. I'll move this default to our metadata and remove this override.